ros2_easy_test/env.py

```diff
@@ -351,3 +393,66 @@ def call_service(
 
         future = client.call_async(request)
         return cast(SrvTypeResponse, self.await_future(future, timeout=timeout_call))
+
+    def _get_action_objects(self, name: str, action_type: Type) -> ActionObjects:
+        """Returns the service client for the given service name."""
+
+        with self._registered_actions_lock:
+            try:
+                return self._registered_actions[name]
+            except KeyError:
+                objects = ActionObjects(
+                    client=ActionClient(self, action_type, name, callback_group=self._action_cb_group)
+                )
+                self._registered_actions[name] = objects
+                return objects
+
+    def send_action_goal_and_wait_for_response(
+        self,
+        name: str,
+        action_type: Type,
+        goal_msg: Any,
+        timeout_availability: Optional[float] = 1,
+        timeout_send_goal: Optional[float] = 1,
+        timeout_get_result: Optional[float] = 10,
+    ) -> tuple[Optional[ClientGoalHandle], Optional[List[Any]], Optional[Any]]:
+        """Sends the goal to the given action and returns the response, feedbacks and result.
+
+        The action type is inferred automatically from the request type.
+
+        Args:
+            name: The name of the service
+            request: The request to send to the service
+            timeout_availability: The timeout to wait for the service to be available
+            timeout_call: The timeout to wait for the service to respond
+
+        Returns:
+            The response of the service
+
+        Raises:
+            TimeoutError: If the service did not respond within the timeout
+        """
+        action_objects = self._get_action_objects(name, action_type)
+
+        is_ready = action_objects.client.wait_for_server(timeout_availability)
+        if not is_ready:
+            raise TimeoutError(f"Action {name} did not become ready within {timeout_availability} seconds")
+
+        action_objects.send_goal_future = action_objects.client.send_goal_async(
+            goal_msg, feedback_callback=action_objects.feedback_callback
+        )
+        action_objects.send_goal_future.add_done_callback(action_objects.goal_response_callback)
+
+        action_objects.goal_handle = self.await_future(
+            action_objects.send_goal_future, timeout=timeout_send_goal
+        )
+
+        # Spin so we have a goal_response_callback and get_result_future is set.
```

Comment:
Ah, okay. Yeah, this is a bit unfortunate, but probably no simple way to avoid this.

I noticed, though, that if the goal is not accepted (i.e., it is rejected), won't this block be forever? I think we should fail the test (`raise AssertionError("...")`) in that case since the action was not completed successfully.

Reply:
How exactly do you want to do that, add another timeout and keep checking the time every spin?

Reply:
Added another timeout.